src/main/java/org/commcare/formplayer/services/SubmitService.java

```diff
@@ -34,6 +35,12 @@
         submitTimer.start();
         try {
             RestTemplate restTemplate = new RestTemplate();
+            restTemplate.setErrorHandler(new DefaultResponseErrorHandler() {
+                @Override
+                protected boolean hasError(HttpStatus statusCode) {
```

Comment:
Were you intending this to only apply to 429s?

I think that would be 

```
if (statusCode == HttpStatus.TOO_MANY_REQUESTS) {
    return false;
} 

return super.hasError(statusCode);
```

Reply:
I think the code as I wrote it here is what I want to do. I kind of put two changes into one commit in retrospect so it's a little confusing. But even aside from the 429 thing, the `if(!submitResponse.getStatusCode().is2xxSuccessful())` bit above wasn't actually catching errors. So this change is to make non-2xx responses get handled by that if clause instead of raising an exception, across the board.

(The second change is then to add sublogic in that handling for the 429 above.)

Reply:
Got it, that makes sense. Thanks for confirming